extendables/channelname.js

```diff
@@ -0,0 +1,34 @@
+const { Extendable } = require('klasa');
+const { GuildChannel } = require('discord.js');
+const channelRegex = new RegExp(/^(?:<#)?(\\d{17,19})>?$/);
+module.exports = class extends Extendable {
+
+	constructor(...args) {
+		super(...args, ['ArgResolver'], { klasa: true });
+	}
+
+	async extend(arg, currentUsage, possible, repeat, msg) {
+		if (arg instanceof GuildChannel) return arg;
+		const matches = channelRegex.exec(arg);
+		if (matches && msg.guild.channels.has(matches[1])) return msg.guild.channels.get(matches[1]) || null;
+		const search = arg.toLowerCase();
+		let channels = msg.guild.channels.filterArray(channelFilterInexact(search));
```

Comment:
Instead of all this, you could have two arrays, inexact and exact, with a for-of loop, and iterate it as God asks.